src/widgets/modals/DemoUserModal.js

```diff
@@ -0,0 +1,243 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2016
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Image, StyleSheet, Text, TextInput, View, TouchableOpacity } from 'react-native';
+import { Button } from 'react-native-ui-components';
+import Modal from 'react-native-modalbox';
+import { DemoSiteRequest } from '../../authentication';
+import globalStyles, { SUSSOL_ORANGE, GREY, WARM_GREY } from '../../globalStyles';
+import { authStrings, generalStrings, demoUserModalStrings } from '../../localization';
+import { ConfirmModal } from '../../widgets';
+import Icon from 'react-native-vector-icons/FontAwesome';
+
+export class DemoUserModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.demoSiteRequest = new DemoSiteRequest();
+    this.state = {
+      status: 'submit', // submit, submitting, submitted, error
+      error: '',
+      username: '',
+      password: '',
+      repeatPassword: '',
+      email: '',
+    };
+    this.errorTimeoutId = null;
+  }
+
+  componentWillUpdate() {
+    if (this.errorTimeoutId) clearTimeout(this.errorTimeoutId);
+  }
+
+  onDemoRequestSubmit = async () => {
+    if (!this.canAttemptSubmit) return false;
+    this.setState({ status: 'submitting' });
```

Comment:
I think since we are using `state.status: 'submitting'` for esthetic purposes are that not that critical, we can let it pass, because there are other instances in the that would do things like this and we we keep nesting we would have few nested codes.